cli/src/commands/operation.rs

```diff
@@ -172,20 +172,27 @@ fn cmd_op_log(
     };
     let with_content_format = LogContentFormat::new(ui, command.settings())?;
 
+    let op_node_template = {
+        let language = OperationTemplateLanguage::new(
+            repo_loader.op_store().root_operation_id(),
+            current_op_id,
+            command.operation_template_extension(),
+        );
```

Comment:
nit: no need to construct `language` twice.

It also applies to the commit template (and is more important since the underlying cache can be shared), but there's no helper function to do so.

Reply:
It might be worth refactoring this at some point, but idk what a really nice way of doing this is. The straightforward options are to either construct language twice, or to let the constructed language live for longer than it needs to.

Doing something like this would work, but feels pretty icky imo
```
let (tmpl1, tmpl2) = {
    let language = ...;
    (parse_template(template_string1), parse_template(template_string2))
};
```
and would require refactoring `parse_commit_template` into something like a `get_commit_template_language` and using parse template separately.

Reply:
It could be
```
let tmpl1;
let tmpl2;
{
    ...
    tmpl1 = parse(..);
    tmpl2 = parse(..);
}
```
(if you prefer.)

I grouped the `language` and parsing variables just because I generally prefer shorter name and scope. It's not super important.

> would require refactoring parse_commit_template into something like a get_commit_template_language and using parse template separately.

Yep. https://github.com/martinvonz/jj/pull/3319/commits/52a400e1581555851766367bff35749a96acc976

Reply:
Oh neat, I completely forgot you can separate declaration and definition like that for non-mut stuff. Stacked another commit with this change on top, can squash if needed later.